dice-game-ui-angular/src/app/services/local-game-storage-service.ts

```diff
@@ -1,33 +1,70 @@
 import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
-import { GameEngine, SimpleGameEngine, Size } from 'engine';
-import { GameStorageService } from './game-storage-service';
+import { GameEngine, SimpleGameEngine } from 'engine';
+import { GameStorageService, SavedGameEngine } from './game-storage-service';
 
 @Injectable()
 export class LocalGameStorageService implements GameStorageService {
-  createGame(gameEngine: GameEngine): Guid {
-    const guid = Guid.create();
-    localStorage.setItem(guid.toString(), gameEngine.fieldSize.width.toString());
-    return guid;
-  }
+  private readonly savedGamesKey = 'saved-games';
+
+  saveGame(gameEngine: GameEngine, guid: Guid = null): Guid {
+    const allSavedGames = this.getAllSavedGames();
+    if (!guid && allSavedGames.length === 0) {
```

Comment:
So, the first game will never be created.

Reply:
Fixed